### PR TITLE
Add schedule-related search attributes

### PR DESCRIPTION
**Pull request by dnr**
**What changed?**
Add four new builtin search attributes to ES schema: two for scheduled workflows, two for the schedule itself.

**Why?**
The ones for the schedule are used by list functionality, the ones for scheduled workflows are to be able to link scheduled workflows to the schedule that started them.

**How did you test it?**
manually ran against local ES

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: schema/elasticsearch/visibility/versioned/v2/reindex.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Comment (alexshtin) [inline schema/elasticsearch/visibility/versioned/v2/reindex.sh:1]: Let's give it more generic name. `upgrade.sh` ?

File: schema/elasticsearch/visibility/versioned/v2/reindex.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+fi
+
+echo "=== Step 1. Add new builtin search attributes to the new index. ==="
+
+case $ES_VERSION in
+    v6) date_type='date'       ; doc_type='/_doc' ;;
+    *)  date_type='date_nanos' ; doc_type=''      ;;
+esac
+
+new_attrs='
```
Comment (alexshtin) [inline schema/elasticsearch/visibility/versioned/v2/reindex.sh:36]:
```suggestion
new_attributes='
```

**Review by alexshtin — APPROVED** [review]

File: schema/elasticsearch/visibility/versioned/v2/reindex.sh (hunk 1 above)
Comment (dnr) [reply schema/elasticsearch/visibility/versioned/v2/reindex.sh:1]: sounds good

File: schema/elasticsearch/visibility/versioned/v2/reindex.sh (hunk 2 above)
Comment (dnr) [reply schema/elasticsearch/visibility/versioned/v2/reindex.sh:36]: actually `new_mapping` seems more accurate

**Pull request merged**